src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Migrate/ProjectJsonToPackageRefMigrator.cs

```diff
@@ -77,6 +74,8 @@ private static async Task MigrateDependenciesAsync(BuildIntegratedNuGetProject p
                 dependencies.AddRange(targetFramework.Dependencies);
             }
 
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Does this really need to be on the UI thread? I don't see any particular method requiring it?

Reply:
The implementation of `AddOrUpdatePackageReferenceAsync`, in the loop below, switches to the UI thread. By switching to the UI thread before the loop, we can make sure we switch to the UI thread once and do all the work.

Otherwise each iteration of the loop, the `await` will continue on the background thread, and then the next iteration of the loop will again switch to the UI thread.

Reply:
Fine by me.